File: Java/1630. Arithmetic Subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> ans = new ArrayList<>();
        for (int i=0; i<l.length; i++) {
            ans.add(checkArithmetic(Arrays.copyOfRange(nums, l[i], r[i]+1)));
        }
        return ans;
    }
    
    private boolean checkArithmetic(int[] arr){
        if (arr.length <= 2) {
            return true;
        }
        Arrays.sort(arr);
        int diff = arr[1] - arr[0];
        for (int i=2; i<arr.length; i++) {
            if (arr[i] - arr[i-1] != diff) {
                return false;
            }
        }
        return true;
    }
    
}

File: Python3/1630. Arithmetic Subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        ans = []
        for i, left in enumerate(l):
            right = r[i]
            sub = nums[left: right+1]
            ans.append(self.checkArithmetic(sub))
        return ans
            
    def checkArithmetic(self, arr):
        if len(arr) <= 2:
            return True
        
        arr.sort()
        diff = arr[1] - arr[0]
        for i in range(2, len(arr)):
            if arr[i] - arr[i-1] != diff:
                return False
        return True
            